src/EnergyPlus/ElectricBaseboardRadiator.cc

```diff
@@ -973,26 +875,21 @@ namespace ElectricBaseboardRadiator {
         // see if the system was even on.  If any average term is non-zero, then
         // one or more of the radiant systems was running.
 
-        // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
-        int BaseboardNum; // DO loop counter for surface index
-
         ElecBaseboardSysOn = false;
 
         // If this was never allocated, then there are no radiant systems in this input file (just RETURN)
-        if (!allocated(state.dataElectBaseboardRad->QBBElecRadSrcAvg)) return;
+        if (state.dataElectBaseboardRad->NumElecBaseboards == 0) return;
 
         // If it was allocated, then we have to check to see if this was running at all...
-        for (BaseboardNum = 1; BaseboardNum <= state.dataElectBaseboardRad->NumElecBaseboards; ++BaseboardNum) {
-            if (state.dataElectBaseboardRad->QBBElecRadSrcAvg(BaseboardNum) != 0.0) {
+        for (int BaseboardNum = 1; BaseboardNum <= state.dataElectBaseboardRad->NumElecBaseboards; ++BaseboardNum) {
+            auto &elecBaseboard = state.dataElectBaseboardRad->ElecBaseboard(BaseboardNum);
+            elecBaseboard.QBBElecRadSource = elecBaseboard.QBBElecRadSrcAvg;
+            if (elecBaseboard.QBBElecRadSrcAvg != 0.0) {
                 ElecBaseboardSysOn = true;
-                break; // DO loop
             }
         }
 
-        state.dataElectBaseboardRad->QBBElecRadSource = state.dataElectBaseboardRad->QBBElecRadSrcAvg;
-
         // QBBElecRadSource has been modified so we need to redistribute gains
-
         DistributeBBElecRadGains(state);
```

Comment:
Note the break was removed so that all baseboards set QBBElecRadSource = QBBElecRadSrcAvg as it did before. I don't expect this change would cause diffs.

Reply:
Hmmm, yeah I guess that's fine.  The whole storing a single value on the `state.dataElectBaseboardRad` state structure just to use as a single transfer mechanism isn't the best, but it's beyond scope to fix that here.